### PR TITLE
Full project

### PR DESCRIPTION
**Pull request by Paul-Vasylenko**

### CODE REVIEW

File: Assets/Scripts/Core/GameLevelInitializer.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+namespace Core
+{
+    public class GameLevelInitializer : MonoBehaviour
+    {
+        [SerializeField] private PlayerEntity _playerEntity;
+        [SerializeField] private GameUIInputView _gameUIInputView;
+
+        private ExternalDevicesInputReader _externalDevicesInputReader;
+        private PlayerSystem _playerSystem;
+        private ProjectUpdater _projectUpdater;
```
Comment (FilonenkoDima) [inline Assets/Scripts/Core/GameLevelInitializer.cs:17]: Field '_projectUpdater' is assigned but its value is never used

File: Assets/Scripts/Core/Services/Updater/ProjectUpdater.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    public class ProjectUpdater : MonoBehaviour, IProjectUpdater
+    {
+        public static IProjectUpdater Instance;
+        public event Action UpdateCalled;
+        public event Action FixedUpdateCalled;
+        public event Action LateUpdateCalled;
+
+        private bool _isPaused;
+
+        public bool IsPaused
```
Comment (FilonenkoDima) [inline Assets/Scripts/Core/Services/Updater/ProjectUpdater.cs:15]: Property 'IsPaused' can be made private

File: Assets/Scripts/Core/Services/Updater/IProjectUpdater.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+
+namespace Core.Services.Updater
+{
+    public interface IProjectUpdater
+    {
+        event Action UpdateCalled;
+        event Action FixedUpdateCalled;
+        event Action LateUpdateCalled;
```
Comment (FilonenkoDima) [inline Assets/Scripts/Core/Services/Updater/IProjectUpdater.cs:9]: Event 'LateUpdateCalled' is never subscribed to

File: Assets/Scripts/Core/Movement/Controller/Jumper.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            IsJumping = true;
+            _rigidbody.AddForce(Vector2.up * _statValueGiver.GetStatValue(StatType.JumpForce));
+        }
+        
+        public void UpdateJump()
+        {
+            if (_rigidbody.velocity.y < 0 && IsGrounded()) // add is grounded
+            {
+                ResetJump();
+                return;
```
Comment (FilonenkoDima) [inline Assets/Scripts/Core/Movement/Controller/Jumper.cs:36]: Redundant control flow jump statement

File: Assets/Scripts/Core/Movement/Controller/HorizontalMover.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+using StatsSystem.Enum;
+using UnityEngine;
+
+namespace Core.Movement.Controller
+{
+    public class HorizontalMover
+    {
+        private readonly Rigidbody2D _rigidbody;
+        private readonly Transform _transform;
+        private readonly HorizontalMovementData _horizontalMovementData;
```
Comment (FilonenkoDima) [inline Assets/Scripts/Core/Movement/Controller/HorizontalMover.cs:13]: Field '_horizontalMovementData' is assigned but its value is never used

File: Assets/Scripts/Core/Animation/UnityAnimatorController.cs [hunk 6]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
```
Comment (FilonenkoDima) [inline Assets/Scripts/Core/Animation/UnityAnimatorController.cs:1]: Using directive is not required by the code and can be safely removed

File: Assets/Scripts/Player/PlayerEntity.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
+using System;
```
Comment (FilonenkoDima) [inline Assets/Scripts/Player/PlayerEntity.cs:1]: Using directive is not required by the code and can be safely removed

File: Assets/Scripts/Player/PlayerEntity.cs [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+using Core.Animation;
+using Core.Enums;
```
Comment (FilonenkoDima) [inline Assets/Scripts/Player/PlayerEntity.cs:3]: Using directive is not required by the code and can be safely removed

File: Assets/Scripts/StatsSystem/StatsController.cs [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+        public StatsController(List<Stat> currentStats)
+        {
+            _currentStats = currentStats;
+            _activeModificators = new List<StatModificator>();
+            ProjectUpdater.Instance.UpdateCalled += OnUpdate;
+        }
+
+        public float GetStatValue(StatType statType) => _currentStats.Find(stat => stat.Type == statType);
+
+        public void ProcessModificator(StatModificator modificator)
```
Comment (FilonenkoDima) [inline Assets/Scripts/StatsSystem/StatsController.cs:24]: Method 'ProcessModificator' can be made private

**Review by FilonenkoDima — APPROVED** [review]
Були певні косметичні помилки. Код супроводжується коментарями автора. Виявлено баг, що персонаж перестає стрибати. Автор дотримується стилю написання коду. Правильна архітектура.

File: Assets/Scripts/Core/GameLevelInitializer.cs (hunk 1 above)
Comment (Paul-Vasylenko) [reply Assets/Scripts/Core/GameLevelInitializer.cs:17]: that is for future :)

File: Assets/Scripts/Core/Services/Updater/IProjectUpdater.cs (hunk 3 above)
Comment (Paul-Vasylenko) [reply Assets/Scripts/Core/Services/Updater/IProjectUpdater.cs:9]: that is for future :)

**Pull request merged**